Orange/widgets/data/owrandomize.py

```diff
@@ -0,0 +1,119 @@
+    shuffle_metas = Setting(0)
+    scope_prop = Setting(80)
+    random_seed = Setting(0)
+    auto_apply = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+
+        # GUI
+        box = gui.vBox(self.controlArea, "Shuffle")
+        self.class_check = gui.checkBox(
+            box, self, "shuffle_class", "Classes",
+            callback=self._shuffle_check_changed)
+        self.attrs_check = gui.checkBox(
+            box, self, "shuffle_attrs", "Features",
+            callback=self._shuffle_check_changed)
+        self.metas_check = gui.checkBox(
+            box, self, "shuffle_metas", "Metas",
+            callback=self._shuffle_check_changed)
+        gui.separator(box, 10, 10)
+        self.replicable_check = gui.checkBox(
+            box, self, "random_seed", "Replicable shuffling",
+            callback=self._shuffle_check_changed)
+
+        box = gui.vBox(self.controlArea, "Scope")
+        hbox = gui.hBox(box)
+        gui.widgetLabel(hbox, "No data")
+        self.scope_slider = gui.hSlider(
+            hbox, self, "scope_prop", minValue=0, maxValue=100, width=140,
+            createLabel=False, callback=self._scope_slider_changed)
+        gui.widgetLabel(hbox, "All data")
+        self.scope_label = gui.widgetLabel(
+            box, "", alignment=Qt.AlignCenter,
+            sizePolicy=(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
+        self._set_scope_label()
+
+        self.apply_button = gui.auto_commit(
+            self.controlArea, self, "auto_apply", "&Apply",
+            box=False, commit=self.apply)
+
+    def _shuffle_check_changed(self):
+        self.apply()
+
+    def _scope_slider_changed(self):
+        self._set_scope_label()
+        self.apply()
+
+    def _set_scope_label(self):
+        self.scope_label.setText("{}%".format(self.scope_prop))
+
+    def set_data(self, data):
+        self.data = data
+        self.apply()
+
+    def apply(self):
+        data = None
+        if self.data:
+            rand_seed = self.random_seed or None
+            _type = Randomize.RandomizeClasses * self.shuffle_class | \
+                    Randomize.RandomizeAttributes * self.shuffle_attrs | \
+                    Randomize.RandomizeMetas * self.shuffle_metas
+            size = int(len(self.data) * self.scope_prop / 100)
+            random.seed(rand_seed)
+            indices = sorted(random.sample(range(len(self.data)), size))
+            randomized = Randomize(_type, rand_seed)(self.data[indices])
+            data = self.data.copy()
+            for i, instance in zip(indices, randomized):
+                data[i] = instance
+        self.send("Data", data)
+
+    def send_report(self):
+        text = "No shuffling"
+        labels = ["Classes", "Features", "Metas"]
+        include = [self.shuffle_class, self.shuffle_attrs, self.shuffle_metas]
+        if sum(include):
+            text = ", ".join([l for i, l in zip(include, labels) if i])
```

Comment:
I prefer printing "a, b, c, d and e", where possible. Try something like

    def send_report(self):
        labels = ["classes", "features", "metas"]
        include = [self.shuffle_class, self.shuffle_attrs, self.shuffle_metas]
        include = [label for labels, i in zip(labels, include)
        text = " and ".join(filter(None, (", ".join(include[:-1]), include[-1:]))) or "none"
        self.report_items ...

This doesn't require `if`. If you find the last line too cryptic, you can get rid of `filter` by handling `len(include) == 1` separately.